src/components/split-button/split-button.component.js

```diff
@@ -17,108 +17,99 @@ const marginPropTypes = filterStyledSystemMarginProps(
 
 const CONTENT_WIDTH_RATIO = 0.75;
 
-class SplitButton extends Component {
-  constructor(props) {
-    super(props);
-    this.buttonLabelId = guid();
-    this.additionalButtons = [];
-    this.listening = false;
-    this.isToggleButtonFocused = false;
-    this.userInputType =
-      "ontouchstart" in document.documentElement ? "touchstart" : "click";
-
-    this.splitButtonNode = React.createRef();
-    this.buttonContainer = React.createRef();
-  }
-
-  state = {
-    showAdditionalButtons: false,
-    minWidth: 0,
-  };
-
-  focusToggleButton = () => {
-    this.isToggleButtonFocused = true;
-    this.showButtons();
-  };
-
-  showButtons = () => {
-    document.addEventListener(this.userInputType, this.handleClickOutside);
-    this.setState({
-      showAdditionalButtons: true,
-      minWidth:
-        CONTENT_WIDTH_RATIO *
-        this.splitButtonNode.current.getBoundingClientRect().width,
-    });
-
-    if (!this.listening) {
-      document.addEventListener("keydown", this.handleKeyDown);
-      this.listening = true;
-    }
-  };
-
-  hideButtons = () => {
-    if (this.isToggleButtonFocused) return;
+const SplitButton = ({
+  align = "left",
+  as = "secondary",
```

Comment:
I'd assign the `secondary` default value to the `buttonType` instead of `as` since the `as` prop is meant to be deprecated here

Reply:
if I'll do it then it will be another breaking change. The `as` prop should be deprecated  in a separate task.